app/models/template_kind.rb

```diff
@@ -6,4 +6,12 @@ class TemplateKind < ActiveRecord::Base
   has_many :os_default_templates
   validates :name, :presence => true, :uniqueness => true
   scoped_search :on => :name
+
+  after_save :clear_cache!
```

Comment:
don't think it's important enough a change to make, the less complex the better, IMHO